src/Microsoft.AspNetCore.Mvc.Core/Formatters/OutputFormatter.cs

```diff
@@ -76,9 +93,9 @@ protected virtual bool CanWriteType(Type type)
                         mediaTypes.Add(mediaType);
                     }
                 }
-
-                return mediaTypes;
             }
+
+            return mediaTypes;
```

Comment:
You should double check this expectation - but I'm wondering if this code has always had the wrong assumptions about what's in `contentType`